### DISCUSSION TITLE
Wrong Isotope usage

### DISCUSSION THREAD

**Discussion by oitconz** [bootscore/bootscore#761] [Q&A]
### Describe the bug

Hi I purchased the isotope plugin yesterday.
The  masonry and other layouts are broken. 

**To see the issue:**
I have setup a brand new Wordpress site. 
I installed BootScore Theme, , Grid, IsoTope, Masonry plugins
I also have CF7 and ACf plugins installed  and Classic Editor
The bootscore theme  is active
I have set up categories of animals with sub of feline / canine. and created posts under various categories. 
I have made a page for each of the isoTope layout types except the woocommerce one. 
There has been no other work on this site or any form of alteration except that listed above. 

**The website can be seen here : http://shaneh45.sg-host.com/**
There is a login created and I can send you details directly. 
The demo pages are all present in the top menu. 
Categories are visible in the right side bar. 

- The equal height overlay also doesnt work.  The grid is not equal height
- The equal height overlay shows text outside the box - flowing over the other items below it ie overflow is not hidden
- The masonry overlay shows masonry but the hover causes content to fall into the images below the one being hovered over - ie overflow is not hidden. 
- The masonry overlay is completely broken

Let me know where i can send login and passwords

**If you want to recreate this on your own system:**

There is a copy of the DB as a mysql dump here http://shaneh45.sg-host.com/booscore-mysql-dump.zip
There is a copy of the plugins folder, themes folder and upload folder here : http://shaneh45.sg-host.com/plugin-upload-themes.zip

### Steps to reproduce

Go to : http://shaneh45.sg-host.com/
Look at the pages shown in the top menu

### Screenshots and Additional Info

Screen shot of masonry attached

![bootscore-screenshot-masonry](https://github.com/bootscore/bootscore/assets/84379773/c31e06d5-a637-4670-b1e7-800e70d15b82)

### Website link

http://shaneh45.sg-host.com

**Comment by oitconz** [comment]
WordPress Info Here : 
`
### wp-core ###

version: 6.5.2
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: /%postname%/
https_status: false
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: production
user_count: 2
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /home/customer/www/shaneh45.sg-host.com/public_html
wordpress_size: 90.92 MB (95336605 bytes)
uploads_path: /home/customer/www/shaneh45.sg-host.com/public_html/wp-content/uploads
uploads_size: 14.07 MB (14752800 bytes)
themes_path: /home/customer/www/shaneh45.sg-host.com/public_html/wp-content/themes
themes_size: 15.34 MB (16081911 bytes)
plugins_path: /home/customer/www/shaneh45.sg-host.com/public_html/wp-content/plugins
plugins_size: 29.12 MB (30531276 bytes)
database_size: 3.33 MB (3489792 bytes)
total_size: 152.77 MB (160192384 bytes)

### wp-active-theme ###

name: bootScore (bootscore-main)
version: 5.3.4
author: bootScore
author_website: https://bootscore.me
parent_theme: none
theme_features: core-block-patterns, widgets-block-editor, automatic-feed-links, title-tag, post-thumbnails, html5, customize-selective-refresh-widgets, menus, widgets
theme_path: /home/customer/www/shaneh45.sg-host.com/public_html/wp-content/themes/bootscore-main
auto_update: Disabled

### wp-themes-inactive (3) ###

Twenty Twenty-Four: version: 1.1, author: the WordPress team, Auto-updates disabled
Twenty Twenty-Three: version: 1.4, author: the WordPress team, Auto-updates disabled
Twenty Twenty-Two: version: 1.7, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (9) ###

Advanced Custom Fields: version: 6.2.9, author: WP Engine, Auto-updates disabled
bs Contact Form 7: version: 5.3.3, author: Bootscore, Auto-updates disabled
bs Grid: version: 5.6.1, author: Bootscore, Auto-updates disabled
bs Isotope: version: 6.2.1, author: Bootscore, Auto-updates disabled
bs Masonry: version: 5.1.0, author: Bootscore, Auto-updates disabled
Classic Editor: version: 1.6.3, author: WordPress Contributors, Auto-updates disabled
Contact Form 7: version: 5.9.3, author: Takayuki Miyoshi, Auto-updates disabled
Security Optimizer: version: 1.4.13, author: SiteGround, Auto-updates disabled
Speed Optimizer: version: 7.5.0, author: SiteGround, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_GD
imagick_module_version: Not available
imagemagick_version: Not available
imagick_version: Not available
file_uploads: 1
post_max_size: 256M
upload_max_filesize: 256M
max_effective_size: 256 MB
max_file_uploads: 20
gd_version: 2.3.3
gd_formats: GIF, JPEG, PNG, WebP, BMP, XPM
ghostscript_version: 10.02.1

### wp-server ###

server_architecture: Linux 6.6.21-MCIclouder470 x86_64
httpd_software: Apache
php_version: 7.4.33 64bit
php_sapi: apache2handler
max_input_variables: 3000
time_limit: 120
memory_limit: 768M
max_input_time: 120
upload_max_filesize: 256M
php_post_max_size: 256M
curl_version: 8.2.1 OpenSSL/3.1.1
suhosin: false
imagick_availability: false
pretty_permalinks: true
htaccess_extra_rules: true
current: 2024-04-30T23:23:25+00:00
utc-time: Tuesday, 30-Apr-24 23:23:25 UTC
server-time: 2024-04-30T23:23:23+00:00

### wp-database ###

extension: mysqli
server_version: 5.7.44-48-log
client_version: mysqlnd 7.4.33
max_allowed_packet: 33554432
max_connections: 500

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /home/customer/www/shaneh45.sg-host.com/public_html/wp-content
WP_PLUGIN_DIR: /home/customer/www/shaneh45.sg-host.com/public_html/wp-content/plugins
WP_MEMORY_LIMIT: 256M
WP_MAX_MEMORY_LIMIT: 768M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: Undefined
WP_DEVELOPMENT_MODE: undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

`

**Comment by crftwrk** [comment]
Hi, 

quickly checked your site.

> The equal height overlay also doesnt work. The grid is not equal height

This works as expected. As the plugin description says, each image must have the same ratio. Plugin does not crop the images.

> The equal height overlay shows text outside the box - flowing over the other items below it ie overflow is not hidden

This works as expected. As the plugin description says, the excerpt should be short as possible. Plugin does not shorten the excerpt. Write a short excerpt in backend edit screen on the right site.

> The masonry overlay is completely broken

It seems that the is an open < tag which is not closed. You can see this because masonry is wrapped and font is broken. This is not in the plugin, it‘s somewhere on your site. I can check this tomorrow, please send admin data here https://bootscore.me/private-info-form/

BTW, Isotope is a commercial plugin and you have access to private support https://bootscore.me/support/

**Comment by crftwrk** [comment]
Inspected the page. The masonry shortcode is wrapped in a code display `<pre>` tag. Of course, this will not work. Use a shortcode block or add shortcode direct to the content like your other Isotopes.

![screenshot-1](https://github.com/bootscore/bootscore/assets/51531217/9115de39-178b-4f49-8fb0-a3adf9b90ff3)

![screenshot-2](https://github.com/bootscore/bootscore/assets/51531217/b0aaa3b6-8119-439b-84a6-cc878a6e0dad)

#### Overlay templates

- Text length must fit to the image. There is not enough space to show a long excerpt. More information about card-overlays https://getbootstrap.com/docs/5.3/components/card/#image-overlays
- Equal-height overlay images must all have the same size to fit them

<img width="451" alt="screenshot-4" src="https://github.com/bootscore/bootscore/assets/51531217/e6cc206a-b744-4a88-b4a2-f58d1d27143a">

<img width="353" alt="screenshot-3" src="https://github.com/bootscore/bootscore/assets/51531217/f99d144d-a8b5-4fff-afee-1c81d4282981">

Close this as completed now, because here are no bugs. Feel free to ask if you have more questions.

**Comment by oitconz** [comment]
Thanks - found the issue. Code copied off your docs pages  came with its own code when pasted into wordpress pages. 
For some reason all the other pages didn't bring that across with them.  I had also tried it in a short code and got a similar issue - but that was missing tax='' paramter in the short code. 
It might be worth adding a check at around line 130 to see if one of the parameters is missing and put that as a warning - as at present missing a parameter like Tax= invokes PHP warnings across the page.

**Comment by oitconz** [comment]
> BTW, Isotope is a commercial plugin and you have access to private support https://bootscore.me/support/

1 - There is no working support link on the isotop or bootscore page. The only link is in the footer of the site, it says support nad it has no anchor tags / is not clickable. 
2 - the only information on your site re support states :
Support / Contribute

We use GitHub to develop our themes and plugins. If you need help, have a question, find a bug or want to contribute, you’re welcome to open a discussion or an issue in the appropriate repository. 

So - your customers have no way to access support except through this discussion on github or similar chat functions.

**Comment by crftwrk** [comment]
Converted this issue to a discussion, because this is really no bug.

> The only link is in the footer of the site, it says support nad it has no anchor tags / is not clickable.

You find in footer area "Support" with a link to support form https://bootscore.me/support/. When you click on it, you come to a page where you can explain your problem and enter admin data.

However, checked your site and fixed it.

- Disabled classic editor plugin, because it seems that this does not support excerpts anymore. Use blocks.
- Removed duplicate `<pre>` wrapped shortcode in masonry page https://shaneh45.sg-host.com/mansonry/
- Added a custom excerpt to all posts which has a good length to the images
- Added new posts "equal-dogs", "equal-cats" with same image size to show how equal-height-overlay works https://shaneh45.sg-host.com/equal-height-overlay/

Everything works as expected and done as its explained in the documentation

- Additionally, updated your theme to Bootscore 6 which has arrived today. Now you are up-to-date. 

Let me know if there is something more

**Comment by oitconz** [comment]
> You find in footer area "Support" with a link to support form https://bootscore.me/support/. When you click on it, you come to a page where you can explain your problem and enter admin data.

**It is broken.**
1 - Go to your site: 

![firefox_2i8AhjjPc9](https://github.com/bootscore/bootscore/assets/84379773/bf03bb45-685c-4248-b068-fb672eed0e62)

2 - Scroll down to support
![firefox_BgwKHGJo9j](https://github.com/bootscore/bootscore/assets/84379773/11a1aeb9-706f-4e4f-aca9-8e130e222c50)

3 - Click it doesnt work - as it is missing an anchor / link. see screen shot below. 
![firefox_HNikv8lufz](https://github.com/bootscore/bootscore/assets/84379773/069ed001-c23d-4343-806c-fcfdac8ccac0)

Looking into the code and reinspecting the page - it would seem support is a sub heading and you are meant to click the link below that. However there is very little difference between the title and the link so it is natural to click on support as the link and think it has failed. Having found the word support -and with no visual clues to show it is a sub heading, then having clicked on it it is natural to look no further - as you have found what you are looking for and that item has failed. UIX / UX issue. 

Also your instructions above say to click on support - which is just the title.  Maybe , the link below the support sub heading ( which is called [Purchased Products](https://bootscore.me/support/) )

HTH

**Comment by oitconz** [comment]
> * Disabled classic editor plugin, because it seems that this does not support excerpts anymore. Use blocks.

NOPE!!!!

I wont be the only one saying that. There is a reason that classic editor is one of the top 2-3 most downloaded plugins by a factor of 50 or more. There is a significant proportion of users who dont use blocks. They suck, are impossible for non techs to use, add spaghetti html, slow pages down and slow your work flow immensely. 

Categories, and standard wordpress page excerpts work with classic editor. the_excerpt() works with classic page editor.  If the overlays are just meant to show excerpts then they should just show excerpts and work out of the box. that is a bug. 

We write our sites with zero blocks - so our clients can self manage their sites. Blocks are not an option. And 50 million plus downloads show I am not alone in this belief. 

Also when i write code and use get_the_excerpt() or the_excerpt()  it works even when using classic editor.

**Comment by oitconz** [comment]
also ... if I change to Twenty 24 theme the plugin stops working. does it only work with bootscore theme? 
![firefox_FINFHQ6Hsy](https://github.com/bootscore/bootscore/assets/84379773/54472e9b-92f5-45ce-946b-27b47798709a)

**Comment by crftwrk** [comment]
> It is broken.

Yeah, this is the heading and below you will find the link to the support for purchased products. This is how all footer menu widgets work. But it's fine to solve your issues here, just wanted to mention that you have email support if you have purchased a plugin.

> NOPE!!!!

Usually there is nothing wrong using the Classic Editor plugin. But when checking your post edit screen, this plugin hides the custom excerpt meta box, so I quickly disabled it. You may check if there are some plugin settings to enable excerpt, because you really need one.

> If the overlays are just meant to show excerpts then they should just show excerpts and work out of the box. that is a bug.

Isotope uses https://getbootstrap.com/docs/5.3/components/card/#image-overlays for overlay templates. Of course, it's physically not possible to show long content inside a 3x2cm image. That's why you have to shorten excerpt. Bootstrap says:  

"_Note that content should not be larger than the height of the image. If content is larger than the image the content will be displayed outside the image._"

Again, this is no bug. I setup your overlay correctly and works fine. 

> also ... if I change to Twenty 24 theme the plugin stops working. does it only work with bootscore theme?

Yes, Isotope is a plugin for Bootscore. It's possible to get them work with other themes but needs some adaption because it uses code that only exists in Bootscore. At least it needs a theme which has Bootstrap 5 installed, Twenty 24 is not a Bootstrap theme.

**Comment by oitconz** [comment]
>Usually there is nothing wrong using the Classic Editor plugin. But 
when checking your post edit screen, this plugin hides the custom 
excerpt meta box, so I quickly disabled it. You may check if there are 
some plugin settings to enable excerpt, because you really need one.


IMHO you shouldnt need to use the custom meta excerpt box. We hide it to 
make life easier for clients - the_excerpt() and get_the_excerpt() dont 
need the meta box ( or at least didnt used to if they do now).  isotope 
should use the excerpt out of the box - not the entire content, if space 
is an issue. Surely that makes sense.

Shane

------ Original Message ------
From "Basti" ***@***.***>
To "bootscore/bootscore" ***@***.***>
Cc "oitconz" ***@***.***>; "Author" 
***@***.***>
Date 4/05/2024 8:24:17 pm
Subject Re: [bootscore/bootscore] Wrong Isotope usage (Discussion #761)

**Comment by crftwrk** [comment]
If no custom excerpt is set, WordPress uses the first 55 words for the excerpt and this is too long for this template. You can use a custom excerpt or shorten entire excerpts by a function, but this may lead into issues with other loop items. 

Bootscore is a theme for developers and follows straight WordPress and Bootstrap with no modifications here. There are endless possibilities making this fit to your project using code.
